misc/scripts/library-coverage/compare-files-comment-pr.py

```diff
@@ -41,15 +46,34 @@ def compare_files_str(file1, file2):
     return ret
 
 
-def comment_pr(folder1, folder2, output_file, repo, pr_number, run_id):
-    compare_folders(folder1, folder2, output_file)
+def write_diff_for_run(output_file, repo, run_id):
+    folder1 = "out_base"
+    folder2 = "out_merge"
+    try:
+        utils.subprocess_run(["gh", "run", "download", "--repo", repo, "--name",
+                              "csv-framework-coverage-base", "--dir", folder1, str(run_id)])
+        utils.subprocess_run(["gh", "run", "download", "--repo", repo, "--name",
+                              "csv-framework-coverage-merge", "--dir", folder2, str(run_id)])
+        utils.subprocess_run(["gh", "run", "download", "--repo", repo, "--name",
+                              "pr", "--dir", "pr", str(run_id)])
+
+        compare_folders(folder1, folder2, output_file)
+    finally:
+        if os.path.isdir(folder1):
+            shutil.rmtree(folder1)
+
+        if os.path.isdir(folder2):
+            shutil.rmtree(folder2)
+
+
+def get_comment_text(output_file, repo, run_id):
     size = os.path.getsize(output_file)
     if size == 0:
         print("No difference in the coverage reports")
         return
 
     comment = ":warning: The head of this PR and the base branch were compared for differences in the framework coverage reports. " + \
-        "The generated reports are available in the [artifacts of this workflow run](https://github.com/" + repo + "/actions/runs/" + run_id + "). " + \
+        "The generated reports are available in the [artifacts of this workflow run](https://github.com/" + repo + "/actions/runs/" + str(run_id) + "). " + \
```

Comment:
Gah this gets me every time. Python 3.9 has `f`-strings for better interpolation, if you prefer.